src/GafferTest/ContextTest.cpp

```diff
@@ -206,6 +207,67 @@ void GafferTest::testEditableScope()
 
 }
 
+// Create the number of contexts specified, and return a count for how many collisions there are
+// in the last 32 bits of the context hash.  The uniformity of the last 32 bits should be
+// representative of the uniformity of the hash as a whole ( we don't want to check for collisions
+// in the whole 128 bit hash, since it would take years for one to occur randomly )
+// "mode" switches betwen 4 modes for creating contexts:
+//   0 :  1 entry with a single increment int
+//   1 :  40 fixed strings, plus a single incrementing int
+//   2 :  20 random floats
+//   3 :  an even mixture of the previous 3 modes
+// "chunk" is set to 0-3 to set which of the four 32 bit chunks of the hash to take, from the lowest 32
+//   bits to the highest 32 bits
+// "seed" can be used to perform different runs to get an average number.
+// The goal is that regardless of how we create the contexts, they are all unique, and should therefore
+// have an identical chance of collisions if our hashing performs ideally.
+int GafferTest::countContextHash32Collisions( int contexts, int mode, int chunk, int seed )
+{
+	std::unordered_set<uint32_t> used;
+
+	srand( seed );
+	int collisions = 0;
+	for( int i = 0; i < contexts; i++ )
+	{
+		int curMode = mode;
+		int element = seed * contexts + i;
```

Comment:
This may be a stupid question, but I assume `element` is the "entry with the single increment int", so why the `seed * contexts` constant offset? To try to keep the values used completely different for each seed?

Reply:
I should probably name it "elementSeed", and yes, the goal is to keep the values used completely different for each seed.  We run this test multiple times to reduce random variation, so that we can expect to see the ideal number of collisions +/- 10%, rather than +/- 50%.  If we used the same values for the different runs, we wouldn't get this variation reduction.